packages/expo-ui/ios/Button/Button.swift

```diff
@@ -3,60 +3,53 @@
 import SwiftUI
 import ExpoModulesCore
 
-struct Button: ExpoSwiftUI.View, ExpoSwiftUI.WithHostingView {
+struct Button: ExpoSwiftUI.View {
   @ObservedObject var props: ButtonProps
-  @EnvironmentObject var shadowNodeProxy: ExpoSwiftUI.ShadowNodeProxy
-
-  init(props: ButtonProps) {
-    self.props = props
-  }
 
   var body: some View {
-    ExpoSwiftUI.AutoSizingStack(shadowNodeProxy: shadowNodeProxy, axis: .both) {
-      SwiftUI.Button(
-        role: props.buttonRole?.toNativeRole(),
-        action: {
-          props.onButtonPressed()
-        },
-        label: {
-          if let systemImage = props.systemImage {
-            Label(props.text, systemImage: systemImage)
-          } else {
-            Text(props.text)
-          }
-        })
-      .disabled(props.disabled)
-      .tint(props.color)
-      // TODO: Maybe there is a way to do a switch statement similarly to the `if` extension?
-      .if(props.variant == .bordered, {
-        $0.buttonStyle(.bordered)
-      })
-      .if(props.variant == .plain, {
-        $0.buttonStyle(.plain)
-      })
-      .if(props.variant == .borderedProminent, {
-        $0.buttonStyle(.borderedProminent)
-      })
-      #if !os(tvOS)
-      .if(props.variant == .borderless, {
-        $0.buttonStyle(.borderless)
-      })
-      #endif
+    SwiftUI.Button(
+      role: props.buttonRole?.toNativeRole(),
+      action: {
+        props.onButtonPressed()
+      },
+      label: {
+        if let systemImage = props.systemImage {
+          Label(props.text, systemImage: systemImage)
```

Comment:
Since we reuse this as a primitive, we should probably allow children here – I can followup in a separate PR